packages/flutter/lib/src/services/binding.dart

```diff
@@ -415,6 +416,13 @@ mixin ServicesBinding on BindingBase, SchedulerBinding {
   void setSystemUiChangeCallback(SystemUiChangeCallback? callback) {
     _systemUiChangeCallback = callback;
   }
+
+  /// Alert the engine that the binding is registered. This instructs the engine to
+  /// register its top level window handler on Windows. This signals that the app
+  /// is able to process "System.requestAppExit" signals from the engine.
+  Future<void> initializationComplete() async {
```

Comment:
Also, why is this async and returning a Future? Should this be returning the future received from invokeMethod?

Reply:
It is public because it needs to be called explicitly from within the test

Reply:
`@visibleForTesting` then? Or likely better: `@protected` and then have the test subclass the binding to make it callable.

Reply:
I am unsure if either will be possible. The method is called from neither within a class nor a test. For the test, it is called from [dev/integration_tests/windows_startup_test/lib/windows.dart](https://github.com/flutter/flutter/blob/a44f9465beb861321b8e062032960cc8677306cf/dev/integration_tests/windows_startup_test/lib/windows.dart#L60), which is not in a class, and is in the `lib` directory.

Reply:
Does the `@visibleForTesting` lint allow for invocations from other methods annotated with the same annotation? If so, perhaps, we just mark `testCanEnableApplicationLifecycle` as well?

Reply:
In normal running cases, it should be called from within the binding class's `initInstances`. However, because that is called at most once when an app is initialized (e.g. `runApp` or `ensureInitialized`), we also call the method from within the test.

Reply:
You should be able to run the test with a custom binding that exposes the `@protected` `initializationComplete` method. We do this in other places, here's am example: https://github.com/flutter/flutter/blob/master/packages/flutter/test/widgets/run_app_async_test.dart

We have to be careful with what we expose on the bindings. There have been instances before where we exposed things that should have been internal and people started calling them. It made it very hard to evolve the bindings properly. So, for this internal method we should find a way to limit visibility somehow.

Reply:
@goderbauer Thanks for the example. However, I am trying to implement something similar in this test, but I am unsure if it is possible to use a subclass of the binding in this case, because it uses the flutter test driver, which in turn initializes a `_DriverBinding`. If I attempt to initialize a custom `TestBinding`, whichever binding is initialized second will throw an error as only one binding can be initialized. And, of course, the `_DriverBinding` is a private class and cannot expose `initializationComplete`.